extensions/src/Smithy.Identity.Abstractions/IIdentity.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace Smithy.Identity.Abstractions
+{
+    /// <summary>
+    /// An identity is an entity within the SDK representing who the customer is.
+    /// For example, the customer's identity could be anonymouus, a token, a public / private key pair, etc...
+    /// </summary>
+    public abstract class IIdentity
+    {
+        /// <summary>
+        /// The time after which this identity will no longer be valid. If this is null, an expiration time
+        /// is not known (for example, when using long-term credentials).
+        /// </summary>
+        public virtual DateTimeOffset? Expiration { get; }
```

Comment:
Since we have generated types like Cognito `Credentials` that are using `DateTime` we need use `DateTime` here. We can't change the return type in the subclasses.

Once we align on `DateTime` make the credential classes in Cognito and SecurityToken use `override` instead of `new`. We can't use `new` in this case because everything in Core will treat the object as an `IIdentity` so it will use the implementation on `IIdentity` instead of the subclass.

The alternative is we change everywhere in the SDK to use `DateTimeOffset`. There are some valid arguments for changing so if you want to have a wider discussion about changing globally to `DateTimeOffset` we can. @boblodgett you had some thoughts about switching to `DateTimeOffset` for  V4. I suspect the change is to big of a breaking change but given we did change already to nullable types maybe it is worth doing.

Reply:
Updated the type to `DateTime` in https://github.com/aws/aws-sdk-net/commit/285e421c304ad6d5c21deb54e6f85ec374226275